cli/parse-options.js

```diff
@@ -34,12 +38,26 @@ function parseOptions (options) {
     config.url = options.url
   }
 
+  PouchDB.plugin(require('pouchdb-mapreduce'))
+
   if (options.dbUrl) {
-    config.db.url = options.dbUrl
-  }
-  if (options.inMemory) {
+    if (!urlParse(options.db.url).auth) {
+      return next(new Error('Authentication details missing from database URL: ' + options.db.url))
```

Comment:
Before, this block was within an asynchronous block that required a callback (`next`). But now, the `parseOptions` has no callback option, instead it returns a new value directly. So instead of doing `return next(...)` here, we want to throw an error, like so

```js
throw new Error('Authentication details missing from database URL: ' + options.db.url)
```

makes sense?